### ISSUE TITLE
[Bug]: Deleted data is sometimes stilled visible 

### ISSUE THREAD

**Issue by yah01** [milvus-io/milvus#24234]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Environment

```markdown
- Milvus version: 2.2.0
- Deployment mode(standalone or cluster):
- MQ type(rocksmq, pulsar or kafka):    
- SDK version(e.g. pymilvus v2.0.0rc2):
- OS(Ubuntu or CentOS): 
- CPU/Memory: 
- GPU: 
- Others:
```

### Current Behavior

https://jenkins.milvus.io:18080/blue/organizations/jenkins/Milvus%20HA%20CI/detail/PR-24212/8/pipeline/

**Comment by xiaofan-luan** [comment]
so this is on 2.2?

**Comment by jiaoew1991** [comment]
/assign @sunby 
/unassign 

Similar to the issues reported by users, let's take a look together.

**Comment by sunby** [comment]
> /assign @sunby /unassign
> 
> Similar to the issues reported by users, let's take a look together.

Yes, trying to reproduce it.

**Comment by xiaofan-luan** [comment]
any progress on reproducing it?

**Comment by sunby** [comment]
> any progress on reproducing it?

No, I can not reproduce it...

**Comment by yah01** [comment]
The fix patch doesn't work, for 2.2.0, the QueryNode consumes all delete records from channel checkpoint, which means a delete record may be applied multiple times, as the storage of delete records is designed be append-only, which will break the binary-search

**Comment by yah01** [comment]
#24849 will fix this for master, later I will cherry-pick it to 2.2.0
